File: app/src/main/java/com/starcut/starflightclient/MessageOpenedResponse.java
package com.starcut.starflightclient;

public class MessageOpenedResponse implements StarFlightResponse
{
	private final Result result;

	MessageOpenedResponse(Result result)
	{
		this.result = result;
	}

	public Result getResult()
	{
		return result;
	}

	public static enum Result
	{
		/**
		 * The message open was recorded successfully
		 */
		OK,

		/**
		 * The opening of the message had already been recorded
		 */
		ALREADY_OPENED;
	}
}

File: app/src/main/java/com/starcut/starflightclient/starflight/callbacks/MessageOpenedResponse.kt
package com.starcut.starflightclient.starflight.callbacks

class MessageOpenedResponse(val result: Result) : StarFlightResponse {

    enum class Result {
        /**
         * The message open was recorded successfully
         */
        OK,

        /**
         * The opening of the message had already been recorded
         */
        ALREADY_OPENED
    }
}
